apps/system/js/sync_manager.js

```diff
@@ -0,0 +1,547 @@
+        return;
+      }
+
+      var args = [];
+      this.getAssertion().then(assertion => {
+        args.push(assertion);
+        return this.getKeys();
+      }).then(keys => {
+        args.push(keys);
+        args.push(collections);
+        this.doSync.apply(this, args);
+      }).catch(error => {
+        console.error('Sync error', error);
+        Service.request('SyncStateMachine:error', ERROR_SYNC_REQUEST);
+      });
+    },
+
+    _handle_fxaEvent: function(event) {
+      if (!event || !event.detail) {
+        console.error('Wrong event');
+        return;
+      }
+
+      if (event.detail.eventName != 'onlogout') {
+        return;
+      }
+
+      // Logging out from Firefox Accounts should disable Sync.
+      Service.request('SyncStateMachine:disable');
+    },
+
+    _handle_killapp: function(event) {
+      // The synchronizer app can be killed while it's handling a sync
+      // request. In that case, we need to record an error and notify
+      // the state machine about it. Otherwise we could end up on a
+      // permanent 'syncing' state.
+      var killedApp = event.detail.origin;
+      navigator.mozApps.getSelf().onsuccess = event => {
+        var app = event.target.result;
+        app.getConnections().then(connections => {
+          connections.forEach(connection => {
+            if (connection.keyword != SYNC_REQUEST_IAC_KEYWORD ||
+                connection.subscriber != killedApp) {
+              return;
+            }
+            Service.request('SyncStateMachine:error', ERROR_SYNC_APP_KILLED);
+          });
+
+        });
+      };
+    },
+
+    /** Helpers **/
+
+    initStore: function() {
+      var promises = [];
+      [SYNC_STATE,
+       SYNC_STATE_ERROR,
+       SYNC_LAST_TIME].forEach(key => {
+        var promise = new Promise(resolve => {
+          asyncStorage.getItem(key, value => {
+            this.debug(key + ': ' + value);
+            this.store.set(key, value);
+            resolve();
+          });
+        });
+        promises.push(promise);
+      });
+      return Promise.all(promises);
+    },
+
+    saveState: function(state) {
+      this.state = state || Service.query('SyncStateMachine.state');
+    },
+
+    registerSyncRequest: function() {
+      return navigator.sync.registrations().then(registrations => {
+        var existingReg;
+        registrations.some(reg => {
+          if (reg.task === SYNC_TASK) {
+            return !!(existingReg = reg);
+          }
+        });
+
+        if (existingReg) {
+          this.debug('Sync request already exists');
+          return Promise.resolve();
+        }
+
+        this.debug('Registering sync request', location.href);
+
+        return navigator.sync.register(SYNC_TASK, {
+          // minInterval is in seconds.
+          // XXX 2 minutes is only for testing purposes. Will default to 1
+          //     hour.
+          minInterval: this.settings.get('sync.scheduler.interval') || 120,
```

Comment:
maybe you can use the same `DEBUG` flag that the `BaseModule#debug` to do this just to make sure we don't forget it later on